src/dtx/dtx_common.c

```diff
@@ -1067,6 +1071,10 @@ dtx_batched_commit_deregister(struct ds_cont_child *cont)
 		if (rc != ABT_SUCCESS) {
 			D_ERROR("ABT_future_create failed for DTX flush on "
 				DF_UUID" %d\n", DP_UUID(cont->sc_uuid), rc);
+			/* Set sc_cos_shutdown, then dtx_free_dbca() will be
+			 * called when the DTX batched commit ULT exits.
+			 */
+			cont->sc_cos_shutdown = 1;
```

Comment:
why it's called 'shutdown'? shouldn't it's named 'deregistered'? Actually using the existing 'sc_closing' should be fine?

Reply:
sc_closing also affect dtx_resync, in theory, that is not related with CoS tree. That is why use new "sc_cos_shutdown".

Reply:
sc_closing is set to 1 only when dtx_batched_commit_deregister() is called on cont close, so that you can just skip the dtx_free_dbca() on batched commit ULT exiting if sc_closing == 0, did I miss anything?

Reply:
oh, this sc_cos_shutdown is set when (rc != ABT_SUCCESS), I missed that.